### DISCUSSION TITLE
How to get full member of channel when using client.getParticipants()

### DISCUSSION THREAD

**Discussion by tiulinh** [gram-js/gramjs#555] [General]
i use the method:  client.getParticipants(nameChannel,{limit:1000}) to get member of channel. But i receive limit 200 member. How to get full member of channel? Thank you!

**Comment by Drun555** [comment]
TLDR: Currently there's no reliable way to export all channel members with GramJS or MTProto API at all. 

I lost all my nerves on this. 

```
const result = await client.invoke(
    new Api.channels.GetParticipants({
      channel: "username",
      filter: new Api.ChannelParticipantsRecent({}),
      offset: 43,
      limit: 100,
      hash: BigInt("-4156887774564"),
    })
  );
```

ChannelParticipantsRecent is limited to last recent 200 users, and we can't do anything about this. The one way to "resolve" this problem is using ChannelParticipantsSearch, which accepts any character as search prompt. I wrote this code:

```
import { Api, TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
import readline from "readline";

import { EOL } from 'os';
import fs from 'fs';
const apiId = 'xx';
const apiHash = "xx";
const stringSession = new StringSession(""); // fill this later with the value from session.save()

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

(async () => {
  console.log("Loading interactive example...");
  const client = new TelegramClient(stringSession, apiId, apiHash, {
    connectionRetries: 5,
  });
  await client.start({
    phoneNumber: async () =>
      new Promise((resolve) =>
        rl.question("Please enter your number: ", resolve)
      ),
    password: async () =>
      new Promise((resolve) =>
        rl.question("Please enter your password: ", resolve)
      ),
    phoneCode: async () =>
      new Promise((resolve) =>
        rl.question("Please enter the code you received: ", resolve)
      ),
    onError: (err) => console.log(err),
  });
  console.log("You should now be connected.");
  console.log(client.session.save()); // Save this string to avoid logging in again

  let currentOffset = 0;
  const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
  const alphabet = [
    '0','1','2','3','4','5','6','7','8','9',
    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
  ];

  let realUsers = {};
  (async function () {
    for (let i = 0; i < alphabet.length; i++) {
      
      const users = await client.invoke(
        new Api.channels.GetParticipants({
          channel: '-1001554642917',
          filter: new Api.ChannelParticipantsSearch({q: alphabet[i]}),
          offset: 0,
          limit: 200,
          hash: BigInt("-4156887774564"),
        })
      );

      // console.log(users);
      users.users.forEach(user => {
        if (user.username) realUsers[user.username] = user.id;
      });

      console.log('wait 1s, ' + alphabet[i] + ' ' + Object.keys(realUsers).length)
      await sleep(1000)
    }

    fs.writeFileSync('./outputPlain.txt', JSON.stringify(realUsers));

  })()
  

   
  // await client.sendMessage("me", { message: "Hello!" });
})();
```



But this shit will make you insane, because there is always creative people like theese ones:
![image](https://github.com/user-attachments/assets/ca213821-38ea-4b07-a0f8-1573d06dcc76)
![image](https://github.com/user-attachments/assets/4d4c5613-151c-4cf9-ba6f-e6e706c290f6)

So you probably get banned before you'll try all this symbols.
